language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/TemplateClass.scala

```diff
@@ -62,109 +62,165 @@ private[inner] object TemplateClass extends StrictLogging {
+        .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+        .returns(className)
+        .addParameters(methodParameters.asJava)
+        .addStatement("$T $L = new $T(contractId)", idClassName, idFieldName, idClassName)
+        .addStatement(
+          "$T $L = $T.fromValue(record$$)",
+          templateClassName,
+          dataFieldName,
+          templateClassName)
+
+    val callParameterNames = Vector(idFieldName, dataFieldName, agreementFieldName) ++ maybeContractKeyClassName
+      .map(_ => contractKeyFieldName)
+      .toList
+    val callParameters = CodeBlock.join(callParameterNames.map(CodeBlock.of(_)).asJava, ", ")
+    spec.addStatement("return new $T($L)", className, callParameters).build()
+  }
+
+  private val emptyOptional = CodeBlock.of("$T.empty()", classOf[Optional[_]])
 
   private[inner] def generateFromIdAndRecordDeprecated(
       className: ClassName,
       templateClassName: ClassName,
-      idClassName: ClassName): MethodSpec =
-    MethodSpec
-      .methodBuilder("fromIdAndRecord")
-      .addAnnotation(classOf[Deprecated])
-      .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
-      .returns(className)
-      .addParameter(classOf[String], "contractId")
-      .addParameter(classOf[javaapi.data.Record], "record$")
-      .addStatement("$T $L = new $T(contractId)", idClassName, idFieldName, idClassName)
-      .addStatement(
-        "$T $L = $T.fromValue(record$$)",
-        templateClassName,
-        dataFieldName,
-        templateClassName)
-      .addStatement(
-        "return new $T($L, $L, $T.empty())",
-        className,
-        idFieldName,
-        dataFieldName,
-        classOf[Optional[_]])
+      idClassName: ClassName,
+      maybeContractKeyClassName: Option[TypeName]): MethodSpec = {
+    val spec =
+      MethodSpec
+        .methodBuilder("fromIdAndRecord")
+        .addAnnotation(classOf[Deprecated])
+        .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+        .returns(className)
+        .addParameter(classOf[String], "contractId")
+        .addParameter(classOf[javaapi.data.Record], "record$")
+        .addStatement("$T $L = new $T(contractId)", idClassName, idFieldName, idClassName)
+        .addStatement(
+          "$T $L = $T.fromValue(record$$)",
+          templateClassName,
+          dataFieldName,
+          templateClassName)
+
+    val callParameters = Vector(
+      CodeBlock.of(idFieldName),
+      CodeBlock.of(dataFieldName),
+      emptyOptional) ++ maybeContractKeyClassName.map(_ => emptyOptional).toList
+
+    spec
+      .addStatement("return new $T($L)", className, CodeBlock.join(callParameters.asJava, ", "))
       .build()
+  }
+
+  private val getContractId = CodeBlock.of("event.getContractId()")
+  private val getArguments = CodeBlock.of("event.getArguments()")
+  private val getAgreementText = CodeBlock.of("event.getAgreementText()")
+  private def getContractKey(t: TypeName) =
+    CodeBlock.of("event.getContractKey().map($T::fromValue)", t)
```

Comment:
Unfortunately this seems to not work as expected when `t` is a `TupleN` for some N.
The `fromValue` functions for tuples expect additional arguments to convert the components.

Reply:
Totally right, fixed by #1743, thanks for the detailed report.